app/services/foreman/plugin.rb

```diff
@@ -241,8 +241,11 @@ def role(name, permissions)
       default_roles[name] = permissions
       return false if pending_migrations || Rails.env.test?
       Role.transaction do
-        role = Role.where(:name => name).first_or_create
-        role.add_permissions!(permissions) if role.permissions.empty?
+        role = Role.find_by :name => name
+        role = Role.create :name => name, :origin => self.id unless role
+        role.ignore_locking do |r|
+          r.add_permissions!(permissions)
```

Comment:
I'd need to double check but this was creating doubled filterings every time the plugin was initialized. See https://github.com/theforeman/foreman/pull/4168/files#r107688522, skipping for existing permission is required I think.